revert 2617212f and 0fbae654
diff --git a/sway/input/seat.c b/sway/input/seat.c
index 83772f52..ebdbd91e 100644
--- a/sway/input/seat.c
+++ b/sway/input/seat.c
@@ -14,6 +14,7 @@
 #include <wlr/types/wlr_tablet_v2.h>
 #include <wlr/types/wlr_touch.h>
 #include <wlr/types/wlr_xcursor_manager.h>
+#include "config.h"
 #include "list.h"
 #include "log.h"
 #include "sway/config.h"
@@ -1071,7 +1072,6 @@ bool seat_is_input_allowed(struct sway_seat *seat,
 
 static void send_unfocus(struct sway_container *con, void *data) {
 	if (con->view) {
-		view_close_popups(con->view);
 		view_set_activated(con->view, false);
 	}
 }
@@ -1136,6 +1136,10 @@ static void seat_set_workspace_focus(struct sway_seat *seat, struct sway_node *n
 	struct sway_workspace *last_workspace = seat_get_focused_workspace(seat);
 
 	if (node == NULL) {
+		// Close any popups on the old focus
+		if (node_is_view(last_focus)) {
+			view_close_popups(last_focus->sway_container->view);
+		}
 		seat_send_unfocus(last_focus, seat);
 		sway_input_method_relay_set_focus(&seat->im_relay, NULL);
 		seat->has_focus = false;
@@ -1224,6 +1228,11 @@ static void seat_set_workspace_focus(struct sway_seat *seat, struct sway_node *n
 		}
 	}
 
+	// Close any popups on the old focus
+	if (last_focus && node_is_view(last_focus)) {
+		view_close_popups(last_focus->sway_container->view);
+	}
+
 	// If urgent, either unset the urgency or start a timer to unset it
 	if (container && container->view && view_is_urgent(container->view) &&
 			!container->view->urgent_timer) {
